benchmarks/bench_gap_es_score.py

```diff
@@ -0,0 +1,344 @@
+                    self.W_,
+                    "kullback-leibler",
+                    square_root=False,
+                )
+                if self._minibatch_convergence(
+                    batch_size=len(idx),
+                    batch_cost=batch_cost,
+                    n_samples=n_samples,
+                    step=n_iter_ * n_batch + i,
+                    n_steps=self.max_iter * n_batch,
+                ):
+                    break
+            else:
+                # only continue if no break occurred
+                continue
+            break
+
+        # Update self.H_dict_ with the learned encoded vectors (activations)
+        self.H_dict_.update(zip(unq_X, unq_H))
+        return self
+
+
+class ModifiedGapEncoder(GapEncoder):
+    fitted_models_: list[ModifiedGapEncoderColumn]
+
+    def _create_column_gap_encoder(self):
+        return ModifiedGapEncoderColumn(
+            ngram_range=self.ngram_range,
+            n_components=self.n_components,
+            analyzer=self.analyzer,
+            gamma_shape_prior=self.gamma_shape_prior,
+            gamma_scale_prior=self.gamma_scale_prior,
+            rho=self.rho,
+            rescale_rho=self.rescale_rho,
+            batch_size=self.batch_size,
+            tol=self.tol,
+            hashing=self.hashing,
+            hashing_n_features=self.hashing_n_features,
+            max_iter=self.max_iter,
+            init=self.init,
+            add_words=self.add_words,
+            random_state=self.random_state,
+            rescale_W=self.rescale_W,
+            max_iter_e_step=self.max_iter_e_step,
+            max_no_improvement=10,
+            verbose=True,
+        )
+
+
+#########################################################
+# Benchmarking accuracy and speed on actual traffic_violations dataset
+#########################################################
+
+benchmark_name = "gap_encoder_benchmark_es_score"
+
+
+@monitor(
+    memory=True,
+    time=True,
+    parametrize={
+        "high_card_feature": [
+            "seqid",
+            "description",
+            "location",
+            "search_reason_for_stop",
+            "state",
+            "charge",
+            "driver_city",
+            "driver_state",
+            "dl_state",
+        ],
+        "max_rows": [5_000, 20_000, 50_000],
+        "modif": [True, False],
+    },
+    save_as=benchmark_name,
+    repeat=2,
+)
+def benchmark(
+    high_card_feature: str,
+    max_rows: int,
+    modif: bool,
+):
+    print(f"Running benchmark")
+    ds = fetch_traffic_violations()
+    X = np.array(ds.X[high_card_feature]).reshape(-1, 1).astype(str)
+    y = ds.y
+    # only keep the first max_rows rows
+    # split the data into train and test
+    X_train, X_test, y_train, y_test = train_test_split(
+        X[:max_rows],
+        y[:max_rows],
+        test_size=0.2,
+    )
+    if not modif:
+        gap = GapEncoder()
+    else:
+        gap = ModifiedGapEncoder(verbose=True, batch_size=512)
```

Comment:
Maybe, to compare them equally, we should have the same batch size on both? Or am I missing something?
```suggestion
    if not modif:
        gap = GapEncoder(batch_size=512)
    else:
        gap = ModifiedGapEncoder(verbose=True, batch_size=512)
```